packages/nimble-components/src/rich-text-viewer/testing/rich-text-viewer.pageobject.ts

```diff
@@ -0,0 +1,51 @@
+import type { RichTextViewer } from '..';
+
+/**
+ * Page object for the `nimble-rich-text-viewer` component.
+ */
+export class RichTextViewerPageObject {
+    public constructor(
+        private readonly richTextViewerElement: RichTextViewer
+    ) {}
+
+    public getFirstChildTagContents(): string {
+        return this.getFirstMarkdownRenderedElement()?.textContent || '';
+    }
+
+    public getLastChildTagContents(): string {
+        return this.getLastMarkdownRenderedElement()?.textContent || '';
+    }
+
+    public getLastChildAttribute(attribute: string): string {
+        return (
+            this.getLastMarkdownRenderedElement()?.getAttribute(attribute) || ''
+        );
+    }
+
+    public getChildTags(): string[] {
```

Comment:
Could we come up with a name that conveys the recursive nature of this function? "Child tags" to me implies that it would return all the direct children but none of their children. But looks to be traversing one branch of the tree. Maybe something like `getFirstChildTags()`? Or if you change it to traverse the entire tree you could use something like `getDecendantTagsDepthFirst()` (or whatever traversal order you choose). But that might require rewriting tests if that returns additional redundant tags.

Reply:
I have changed the function to traverse the entire tree in a breadth-first manner and so I renamed the function name to `getDescendantTagsBreadthFirst()` and updated the test cases accordingly.

Reply:
As we enable new markdown features we'll have to keep an eye on whether these tests become brittle. I'm a bit worried that we're now over-asserting about the DOM structure and that tests will start failing if the structure changes in innocuous ways, but that might not be a valid concern. If it does become a problem we can always modify our asserts (for example, assert that `getDescendantTagsBreadthFirst()` returns an array that contains two `LI` elements without asserting what else is in the array).

I think what you have is an ok direction for now, let's just keep an eye on how the tests need to be changed in the future.

Reply:
As mentioned in the [other comment](https://github.com/ni/nimble/pull/1335#discussion_r1268750605) I have updated the helper methods and test cases accordingly to assert in a document order instead of a breadth-first manner. Since we changed it to a document order, I believe these tests will not be a major concern even if we add new features like hyperlinks or mention support, those marks will reside in line and I hope these test cases will not be affected. Anyway, we will keep an eye on it and modify the assert when new features are added in the future.